assets/js/translator.js

```diff
@@ -749,50 +753,54 @@ function detectLanguage() {
         textTranslateRequest.abort();
     }
 
-    textTranslateRequest = $.jsonp({
-        url: config.APY_URL + '/identifyLang',
-        beforeSend: ajaxSend,
-        complete: function () {
-            ajaxComplete();
-            textTranslateRequest = undefined;
-        },
+    textTranslateRequest = callApy({
         data: {
             'q': originalText
         },
-        success: function (data) {
-            var possibleLanguages = [];
-            for(var lang in data) {
-                possibleLanguages.push([lang.indexOf('-') !== -1 ? lang.split('-')[0] : lang, data[lang]]);
-            }
-            possibleLanguages.sort(function (a, b) {
-                return b[1] - a[1];
-            });
+        success: detectLanguageSuccessResponse,
+        error: detectLanguageErrorResponse,
+        complete: function () {
+            ajaxComplete();
+            textTranslateRequest = undefined;
+        }
+    }, '/identifyLang');
 
-            var oldSrcLangs = recentSrcLangs;
-            recentSrcLangs = [];
-            for(var i = 0; i < possibleLanguages.length; i++) {
-                if(recentSrcLangs.length < TRANSLATION_LIST_BUTTONS && possibleLanguages[i][0] in pairs) {
-                    recentSrcLangs.push(possibleLanguages[i][0]);
-                }
-            }
-            recentSrcLangs = recentSrcLangs.concat(oldSrcLangs);
-            if(recentSrcLangs.length > TRANSLATION_LIST_BUTTONS) {
-                recentSrcLangs = recentSrcLangs.slice(0, TRANSLATION_LIST_BUTTONS);
-            }
+    return textTranslateRequest;
+}
 
-            curSrcLang = recentSrcLangs[0];
-            $('#srcLangSelect').val(curSrcLang);
-            muteLanguages();
+function detectLanguageSuccessResponse(data) {
```

Comment:
Let's move both of these functions out of global scope and into the `detectLanguage` function? Also let's go with consistent naming here mirroring `success: handleTranslateSuccessResponse,`.